test/tools/router_check/router_check.cc

```diff
@@ -0,0 +1,27 @@
+#include "router.h"
+#include "test/precompiled/precompiled_test.h"
+
+int main(int argc, char* argv[]) {
+
+  if (argc < 3 || argc > 4) {
+    return -1;
+  }
+
+  RouterCheckTool checktool;
+
+  if (!checktool.create(argv[1])) {
+    return -1;
+  }
+
+  // TODO(hennna): Switch to gflags
+  if (argc == 4 && std::string(argv[3]) == "--details") {
+    checktool.showDetails();
+  }
+
+  checktool.compareEntriesInJson(argv[2]);
+
+  // Print out total matches and conflicts
+  std::cout << "Total Y:" << checktool.getYes() << " N:" << checktool.getNo() << std::endl;
```

Comment:
I would simplify and get rid of the total Y/N. In the simple, automated case, just exit 0/1 depending on whether there is a match. If --details is specified, print the details. It would simplify the interface and the implementation as well. No need for the calls to `increaseCounters` in various places.